employees/views.py

```diff
@@ -140,94 +154,118 @@ def order_by_unpaid_salaries(name_filter, descending):
     ).order_by('%sunpaid_salaries' % (descending))
     return query
 
+
+class UpdateSession(View):
+
+    def post(self, request):
+
+        if not request.is_ajax() or not request.method == 'POST':
+            from django.http import HttpResponseNotAllowed
+            return HttpResponseNotAllowed(['POST'])
+        try:
+            order = self.request.POST['order']
+            request.session['order'] = order
+        except KeyError:
+            request.session['order'] = 'last_name'
+
+        position_sale = self.__make_true_or_false_from_POST_request(request, 'position_sale')
+        request.session['position_sale'] = position_sale
+        position_other = self.__make_true_or_false_from_POST_request(request, 'position_other')
+        request.session['position_other'] = position_other
+        position_production = self.__make_true_or_false_from_POST_request(request, 'position_production')
+        request.session['position_production'] = position_production
+        hide_paid_employees_filter = self.__make_true_or_false_from_POST_request(request, 'hide_paid_employees_filter')
+        request.session['hide_paid_employees_filter'] = hide_paid_employees_filter
+        hide_zero_salary_months = self.__make_true_or_false_from_POST_request(request, 'hide_zero_salary_months')
+        request.session['hide_zero_salary_months'] = hide_zero_salary_months
+        former_employees = self.__make_true_or_false_from_POST_request(request, 'former_employees')
+        request.session['former_employees'] = former_employees
+        current_employees = self.__make_true_or_false_from_POST_request(request, 'current_employees')
+        request.session['current_employees'] = current_employees
+
+        try:
+            per_page = self.request.POST['per_page']
+            request.session['per_page'] = per_page
+        except:
+            request.session['per_page'] = 10
+
+        try:
+            page = self.request.POST['page']
+            request.session['page'] = page
+        except:
+            request.session['page'] = 1
+        return HttpResponse('ok')
+
+    @staticmethod
+    def __make_true_or_false_from_POST_request(request, filer_name):
+        try:
+            value = request.POST[filer_name]
+            if value == 'on':
+                return True
+            else:
+                return False
+        except:
```

Comment:
Same here and to much logic inside `try:` block. You should limit yourself to single line where You think exception could appear.